multiworld/envs/mujoco/sawyer_reach_torque/sawyer_reach_torque_env.py

```diff
@@ -0,0 +1,358 @@
+
+    def set_goal_xyz(self, pos):
+        qpos = self.data.qpos.flat.copy()
+        qvel = self.data.qvel.flat.copy()
+        qpos[7:10] = pos.copy()
+        qvel[7:10] = [0, 0, 0]
+        self.set_state(qpos, qvel)
+
+    @property
+    def init_angles(self):
+        return [
+            1.02866769e+00, - 6.95207647e-01, 4.22932911e-01,
+            1.76670458e+00, - 5.69637604e-01, 6.24117280e-01,
+            3.53404635e+00,
+            1.07586388e-02, 6.62018003e-01, 2.09936716e-02,
+            1.00000000e+00, 3.76632959e-14, 1.36837913e-11, 1.56567415e-23
+        ]
+
+    @property
+    def goal_dim(self):
+        return 3
+
+    @property
+    def endeff_id(self):
+        return self.model.body_names.index('leftclaw')
+
+    @property
+    def goal_id(self):
+        return self.model.body_names.index('goal')
+
+    def get_diagnostics(self, paths, prefix=''):
+        statistics = OrderedDict()
+        for stat_name in [
+            'hand_distance',
+            'hand_success',
+        ]:
+            stat_name = stat_name
+            stat = get_stat_in_paths(paths, 'env_infos', stat_name)
+            statistics.update(create_stats_ordered_dict(
+                '%s%s' % (prefix, stat_name),
+                stat,
+                always_show_all_stats=True,
+                ))
+            statistics.update(create_stats_ordered_dict(
+                'Final %s%s' % (prefix, stat_name),
+                [s[-1] for s in stat],
+                always_show_all_stats=True,
+                ))
+        return statistics
+
+    """
+    Multitask functions
+    """
+    @property
+    def goal_dim(self) -> int:
+        return 3
+
+    def get_goal(self):
+        return {
+            'desired_goal': self._state_goal,
+            'state_desired_goal': self._state_goal,
+        }
+
+    def sample_goals(self, batch_size):
+        if self.fix_goal:
+            goals = np.repeat(
+                self.fixed_goal.copy()[None],
+                batch_size,
+                0
+            )
+        else:
+            goals = np.random.uniform(
+                self.goal_space.low,
+                self.goal_space.high,
+                size=(batch_size, self.goal_space.low.size),
+            )
+        return {
+            'desired_goal': goals,
+            'state_desired_goal': goals,
+        }
+
+    def compute_rewards(self, actions, obs):
+        achieved_goals = obs['achieved_goal']
+        desired_goals = obs['desired_goal']
+        hand_pos = achieved_goals
+        goals = desired_goals
+        distances = np.linalg.norm(hand_pos - goals, axis=1)
+        if self.reward_type == 'hand_distance':
+            r = -distances
+        elif self.reward_type == 'hand_success':
+            r = -(distances < self.indicator_threshold).astype(float)
+        else:
+            raise NotImplementedError("Invalid/no reward type.")
+        return r
+
+    def set_to_goal(self, goal):
+        pass
```

Comment:
Can you `raise NotImplementedError()`? The reason is that this will prevent bugs in the future where someone assumes that this function does something, but it silently doesn't do what it's supposed to do.

Reply:
Yes will do.